### DISCUSSION TITLE
Increment value

### DISCUSSION THREAD

**Discussion by macktns** [GodotNuts/GodotFirebase#148] [Q&A] [answered]
Is there a way to increment a field value in either the firestore or realtime database? I know that the firestore has an increment method but it is on a FieldValue and I can't figure out how to make it work. https://firebase.google.com/docs/reference/js/firebase.firestore.FieldValue#increment

**Comment by BearDooks** [comment, marked as answer]
@macktns 

We are limited with the methods and functions that Firebase offers in the REST API. I looked again but the increment method is not available to us.

 In Firestore you can use the get() function to pull that data and, increment it, and then us update(). It adds a few steps but it will work. I just updated our Wiki yesterday with the Update function for Firestore. 

https://github.com/GodotNuts/GodotFirebase/wiki/Firestore#update-a-document

I would have to ask some of the other members of the team about the RTD.

**Reply by macktns** [reply]
OK that is what I ended up doing and it works well. I was concerned about the possibility of two people incrementing the same value within quick succession (making a 'likes' field), but I suppose it won't happen much or matter a ton if it does happen.

**Reply by WolfgangSenff** [reply]
So that can be a problem, and technically we can find a way to work around it, but it's surprisingly complicated. I will look into this if you want me to - if you do, just open an issue. I'll know what you're talking about, but feel free to link here as well.

**Reply by WolfgangSenff** [reply]
Actually, I went ahead and added the issue myself. Work on it will be tracked at: #149 .

**Reply by macktns** [reply]
Hey thank you, I think its a really great feature to have that has lots of uses.

**Comment by macktns** [comment]
Just wanted to share, a long time later, that I implemented this by writing a firebase function that increments and just call that function.

**Reply by WolfgangSenff** [reply]
We'll add it eventually once I have time to implement it, but your approach is pretty good too!
